python/tk_framework_desktopserver/shotgun/api_v2.py

```diff
@@ -237,7 +238,16 @@ def get_actions(self, data):
                 # descriptor uri. We can get the descriptor from the toolkit
                 # manager and pass that through along with the entity type from SG
                 # to the core hook that computes the hash.
-                pc_descriptor = pc_data["descriptor"]
+                pc_descriptor = pipeline_config["descriptor"]
+
+                # Since the config entity is going to be passed up as part of the
+                # reply to the client, we need to filter out the descriptor object.
+                # It's neither useful to the client, nor json encodable. We're
+                # copying the dict first, since the original is stored in an
+                # in-memory cache and is likely to be reused.
+                pipeline_config = copy.copy(pipeline_config)
```

Comment:
Wouldn't it be simpler if the in-memory cache was the one who cloned it's results instead? This way you wouldn't have to worry about people corrupting the cache. I doubt that's going to be a performance issue either.